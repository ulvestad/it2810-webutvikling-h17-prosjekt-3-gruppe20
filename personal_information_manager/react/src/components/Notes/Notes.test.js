import React from 'react';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { expect } from 'chai';

import Notes from './Notes';

const jestExpect = global.expect;

it('renders without crashing', () => {
  shallow(<Notes />);
});

it('renders correctly', () => {
  const tree = renderer.create(<Notes />).toJSON();
  jestExpect(tree).toMatchSnapshot();
});

it('opens modal when triggered', () => {
  const wrapper = shallow(<Notes />);

  wrapper.instance().openModal();

  expect(wrapper.state()).to.have.property('modalOpen', true);
});

it('closes modal when triggered', () => {
  const wrapper = shallow(<Notes />);

  wrapper.instance().closeModal();

  expect(wrapper.state()).to.have.property('modalOpen', false);
});
