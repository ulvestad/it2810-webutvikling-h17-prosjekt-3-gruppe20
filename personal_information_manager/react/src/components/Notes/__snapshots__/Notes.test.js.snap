// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="notes"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-8 mx-auto"
      >
        <h2>
          Notes
        </h2>
        <blockquote
          className="blockquote"
        >
          <p
            className="mb-0"
          >
            He listens well who takes notes
          </p>
          <footer
            className="blockquote-footer"
          >
            <cite>
              Dante Alighieri
            </cite>
          </footer>
        </blockquote>
        <div
          className="notesContainer"
        >
          <div
            className="panel panel-default note add-note"
            onClick={[Function]}
          >
            <div
              className="panel-body"
            >
              <i
                aria-hidden="true"
                className="fa fa-plus add"
              />
            </div>
          </div>
           
        </div>
      </div>
    </div>
  </div>
</section>
`;
